### PR TITLE
Trail Map: clean up the plans, features, and feature group map data

### PR DESCRIPTION
**Pull request by southp**
Related to #89692 

## Proposed Changes

This PR cleans up the data introduced in `plans-list`, `features-list`, `feature-group-map` and the related function in `experiments`. 

## Why are these changes being made?

Per announcement made in pau2Xa-5Ol-p2#comment-15735, the trail map language offering test has been concluded as rolling out the control variation. We should also clean up the data side in ease of future maintenance.

## Testing Instructions

* Mainly code review to make sure the clean-up makes sense.
* The logged-in /plans and /start/plans should look and work as it used to be.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/calypso-products/src/features-list.tsx [hunk 1]
```diff
@@ -2156,10 +2087,7 @@ const FEATURES_LIST: FeatureList = {
 	},
 	[ FEATURE_MIN_MAX_ORDER_QUANTITY ]: {
 		getSlug: () => FEATURE_MIN_MAX_ORDER_QUANTITY,
-		getTitle: () =>
-			isTrailMapVariant()
-				? i18n.translate( 'Min/max order quantities' )
-				: i18n.translate( 'Min/max order quantity' ),
+		getTitle: () => i18n.translate( 'Min/max order quantities' ),
```
Comment (aneeshd16) [inline packages/calypso-products/src/features-list.tsx:2090]:
```suggestion
		getTitle: () => i18n.translate( 'Min/max order quantity' ),
```

Small fix. :)

**Review by aneeshd16 — COMMENTED** [review]
Nice cleanup! Can you please also remove the trail map storybook versions?

**Comment by southp** [comment]
@aneeshd16 
> Can you please also remove the trail map storybook versions?

Yes, actually I'm now working on a follow-up PR for that :)

File: packages/calypso-products/src/features-list.tsx (hunk 1 above)
Comment (southp) [reply packages/calypso-products/src/features-list.tsx:2090]: Oops. Thanks for catching it :)

**Review by aneeshd16 — APPROVED** [review]

**Comment by southp** [comment]
To close the loop here, here is the storybook clean-up: https://github.com/Automattic/wp-calypso/pull/92159

**Pull request merged**